docs/api/utils/proxyWithComputed.mdx

```diff
@@ -48,16 +36,60 @@ const state2 = proxyWithComputed(
 )
 
 // if you want a computed value to derive from another computed, you must declare the dependency first:
-const state = proxyWithComputed(
-  {
-    count: 1,
-  },
+const state3 = proxyWithComputed(
+  { count: 1 },
   {
     doubled: memoize((snap) => snap.count * 2),
     quadrupled: memoize((snap) => snap.doubled * 2),
   }
 )
 ```
 
-The last use case fails to infer types in TypeScript
-[#192](https://github.com/pmndrs/valtio/issues/192).
+The last use case fails to infer types in TypeScript, see [#192](https://github.com/pmndrs/valtio/issues/192).
+
+#### Disclaimer on `proxy-memoize`
+
+Be careful not to overuse `proxy-memoize` because `proxy-memoize` and `useSnapshot` do similar optimization and double optimization may lead to less performance.
+
+The similarity is that both `proxy-memoize` and `useSnapshot` keep caches of data; the difference is that getters invoked against `snapshot`s or `useSnapshot`s are not cached and will re-invoke the getter/re-calculate the result each time they are accessed; whereas `proxy-memoize` will use the `snap` argument to calculate the getter result only once per snapshot. )
+
+#### Compared to Classes
+
+Using `proxyWithComputed` is similar in spirit to using classes, e.g. the above `proxyWithComputed` could be written as:
+
+```js
+class Author {
+  firstName = 'Alec'
+  lastName = 'Baldwin'
+  get fullName() {
+    return this.firstName + ' ' + this.lastName
+  }
+  set fullName(value: string) {
+    ;[this.firstName, this.lastName] = value.split(' ')
+  }
+}
+const state = proxy(new Author())
+```
+
+The major difference is that:
+
+- `proxyWithComputed` allows using `memoize` to cache getters with course-grained dependency tracking
+
+  "Course-grained" insofar as the getter will be recalculated whenever any field in the snapshot changes, i.e. it has a version bump due to any mutation. This in contract with `useSnapshot`, which implements "fine-grained" dependency tracking based on the fields specifically accessed within the JSX.
+
+#### Compared to object getters
```

Comment:
Same here.

I would like to have a comparison between classes and object getters.